# Data analysis
## Import libraries

In [4]:
import pandas as pd
import os

datadir = "../data/original"

## Data Inspection
### ABS_Population_and_people_by_2021_SA2_(Beta).csv
#### Dataset:

In [5]:
data = pd.read_csv(os.path.join(datadir, "ABS_Population_and_people_by_2021_SA2_(Beta).csv"))
data.head()

OBJECTID SA2_CODE_2021                    SA2_NAME_2021  AREA_ALBERS_SQKM  \
0         1     101021007                        Braidwood         3418.3525   
1         2     101021008                          Karabar            6.9825   
2         3     101021009                       Queanbeyan            4.7620   
3         4     101021010                Queanbeyan - East           13.0032   
4         5     101021012  Queanbeyan West - Jerrabomberra           13.6748   

                                  ASGS_LOCI_URI_2021  ERP_P_202021  \
0  https://linked.data.gov.au/dataset/asgsed3/SA2...        4330.0   
1  https://linked.data.gov.au/dataset/asgsed3/SA2...        8546.0   
2  https://linked.data.gov.au/dataset/asgsed3/SA2...       11370.0   
3  https://linked.data.gov.au/dataset/asgsed3/SA2...        5093.0   
4  https://linked.data.gov.au/dataset/asgsed3/SA2...       12743.0   

   ERP_212021  ERP_M_202021  ERP_F_202021  ERP_192021  ...  CENSUS_302021  \
0         1.3        2248.0        2082.0        49.9  ...            3.5   
1      1223.9        4324.0        4222.0        36.3  ...            5.2   
2      2387.7        5788.0        5582.0        36.3  ...           14.2   
3       391.7        2671.0        2422.0        36.2  ...           11.9   
4       931.9        6387.0        6356.0        36.6  ...            4.1   

   CENSUS_312021  CENSUS_392021  CENSUS_332021  ADFS_22021  ADFS_32021  \
0            7.8          180.0            4.1        20.0       182.0   
1            5.7         1228.0           14.4       109.0       257.0   
2            4.9         2615.0           23.1       150.0       354.0   
3            4.5         1037.0           20.4       152.0       197.0   
4            3.0         1597.0           12.5       585.0       587.0   

   ADFS_42021  ADFS_52021  SHAPE_Length  SHAPE_Area  
0         0.5         5.0      3.913695    0.339397  
1         1.6         3.7      0.138642    0.000693  
2         1.6         3.7      0.107150    0.000472  
3         3.5         4.6      0.189549    0.001290  
4         5.8         5.9      0.193368    0.001356  

[5 rows x 161 columns]

#### Notes:
Contains population data for each SA2 region in Australia. The data is from the Australian Bureau of Statistics (ABS) and is based on the 2021 Census. The data is in a CSV file format.
SA2 code column format: 101021007 (SA2_CODE_2021)

### duration_sa2_hospital_shortest.pkl
#### Dataset:

In [20]:
data = pd.read_pickle(os.path.join(datadir, "duration_sa2_hospital_shortest.pkl"))
data.head()

SA2_5DIG16  shortest_time_sec  shortest_time_min
0       11007        1143.589117               20.0
1       11008         374.559538                7.0
2       11009         202.835640                4.0
3       11010         272.384746                5.0
4       11011        1693.438621               29.0

#### Notes:
This is the shortest travel time between Statistical Areas Level 2 (SA2s) and any hospital in
Australia. However, the hospital locations are not provided in this dataset, only the travel times.

SA2 code column format: 11007 (SA2_5DIG16)

NOTE: This is also available for the Mesh Block level, which is a more granular level than SA2.

### duration_mb_hospitals.pkl
#### Dataset:

In [7]:
data = pd.read_pickle(os.path.join(datadir, "duration_mb_hospitals.pkl"))
data.head()

MB_CODE16  SA2_5DIG16      xcoord     ycoord  Person  time_to_1  \
0  80000009499       89499         NaN        NaN     538        NaN   
1  80000010000       81123  149.082275 -35.291847       0        NaN   
2  80000020000       81123  149.098221 -35.272163       0        NaN   
3  80000030000       81049  149.101807 -35.288490       0        NaN   
4  80000040000       81049  149.119354 -35.292336       0        NaN   

   time_to_2  time_to_3  time_to_4  time_to_5  ...  time_to_1002  \
0        NaN        NaN        NaN        NaN  ...           NaN   
1        NaN        NaN        NaN        NaN  ...           NaN   
2        NaN        NaN        NaN        NaN  ...           NaN   
3        NaN        NaN        NaN        NaN  ...           NaN   
4        NaN        NaN        NaN        NaN  ...           NaN   

   time_to_1003  time_to_1004  time_to_1005  time_to_1006  time_to_1007  \
0           NaN           NaN           NaN           NaN           NaN   
1           NaN           NaN           NaN           NaN           NaN   
2           NaN           NaN           NaN           NaN           NaN   
3           NaN           NaN           NaN           NaN           NaN   
4           NaN           NaN           NaN           NaN           NaN   

   time_to_1008  time_to_1009  time_to_1010  time_to_1011  
0           NaN           NaN           NaN           NaN  
1           NaN           NaN           NaN           NaN  
2           NaN           NaN           NaN           NaN  
3           NaN           NaN           NaN           NaN  
4           NaN           NaN           NaN           NaN  

[5 rows x 1016 columns]

#### Notes:
This is the shortest travel time between Mesh Blocks (MBs) and any hospital in Australia.

SA2 code column format: 89499 (SA2_5DIG16)
MB code column format: 80000009499 (MB_CODE16)

time_to_X has X as hospital ID

NOTE: This is also available for the SA2 level, which is a less granular level than MB.

CONTAINS A LOT OF NAN VALUES, SA2 IS BETTER


### ed_admissions.csv
#### Dataset:

In [8]:
data = pd.read_csv(os.path.join(datadir, "ed_admissions.csv"), encoding="latin1")
data.head()

SA Health Admissions from Emergency Department (Admit To Ward/EECU) for public metropolitan and major country hospitals by quarter  \
0                                                NaN                                                                                   
1                                       LHN/Hospital                                                                                   
2                                                NaN                                                                                   
3              Central Adelaide Local Health Network                                                                                   
4                                                NaN                                                                                   

  Unnamed: 1                    Unnamed: 2   Unnamed: 3   Unnamed: 4  \
0        NaN                           NaN          NaN          NaN   
1        NaN                           NaN      2012-13          NaN   
2        NaN                           NaN  Sep Quarter  Dec Quarter   
3        NaN                           NaN          NaN          NaN   
4       0014  The Queen Elizabeth Hospital        3,375        3,325   

    Unnamed: 5   Unnamed: 6   Unnamed: 7   Unnamed: 8   Unnamed: 9  ...  \
0          NaN          NaN          NaN          NaN          NaN  ...   
1          NaN          NaN      2013-14          NaN          NaN  ...   
2  Mar Quarter  Jun Quarter  Sep Quarter  Dec Quarter  Mar Quarter  ...   
3          NaN          NaN          NaN          NaN          NaN  ...   
4        3,258        3,327        3,356        3,366        3,347  ...   

   Unnamed: 17  Unnamed: 18  Unnamed: 19  Unnamed: 20  Unnamed: 21  \
0          NaN          NaN          NaN          NaN          NaN   
1          NaN          NaN      2016-17          NaN          NaN   
2  Mar Quarter  Jun Quarter  Sep Quarter  Dec Quarter  Mar Quarter   
3          NaN          NaN          NaN          NaN          NaN   
4        3,369        3,382        3,390        3,507        3,267   

   Unnamed: 22  Unnamed: 23  Unnamed: 24  Unnamed: 25  Unnamed: 26  
0          NaN          NaN          NaN          NaN          NaN  
1          NaN      2017-18          NaN          NaN          NaN  
2  Jun Quarter  Sep Quarter  Dec Quarter  Mar Quarter  Jun Quarter  
3          NaN          NaN          NaN          NaN          NaN  
4        3,223        3,673        3,963        3,999        3,854  

[5 rows x 27 columns]

#### Notes:
File needs manual cleaning, the table is pasted in there in a non-csv format.
It contains the number of emergency department admissions for each SA2 region in Australia. (patients gaining admission to a hospital through the emergency department)

### myhospitals-contact-details.csv

#### Dataset:

In [25]:
data = pd.read_csv(os.path.join(datadir, "myhospitals-contact-details.csv"), encoding="latin1")
print(data.shape)
data.head()

(1011, 15)


Hospital name  Phone number          Street address  \
0   Abbotsford Private Hospital  08 9200 6282     61 Cambridge Street   
1               Adelaide Clinic  08 8269 8100         33 Park Terrace   
2  Adelaide Day Surgery Pty Ltd  08 8239 4900        18 North Terrace   
3   Adelaide Eye & Laser Centre  08 8274 7000      215 Greenhill Road   
4          Adelaide Surgicentre  08 8211 0000  89 King William Street   

             Suburb  Postcode State Local Hospital Network (LHN)  \
0  West Leederville      6007    WA                          NaN   
1         Gilberton      5081    SA                          NaN   
2          Adelaide      5000    SA                          NaN   
3          Eastwood      5063    SA                          NaN   
4         Kent Town      5067    SA                          NaN   

  Primary Health Network area (PHN)  \
0                               NaN   
1                               NaN   
2                               NaN   
3                               NaN   
4                               NaN   

                                            Website Description   Sector  \
0                     www.abbotsfordhospital.com.au         NaN  Private   
1                        www.adelaideclinic.com.au/         NaN  Private   
2  http://www.curagroup.com.au/adelaide-day-surgery         NaN  Private   
3                                   www.aelc.com.au         NaN  Private   
4                                   www.asec.net.au         NaN  Private   

    Beds   Latitude   Longitude  Hospital_ID  
0    <50 -31.940992  115.837344            1  
1  50-99 -34.906580  138.613167            2  
2    <50 -34.922409  138.588817            3  
3    NaN -34.940420  138.621500            4  
4    <50 -34.917691  138.621022            5

#### Notes:
Metadata for hospitals in Australia. Contains hospital name, address, phone number, and website.
Also contains hospital ID used in the duration_mb_hospitals.pkl dataset. (time_to_X has X as hospital ID, stored in Hospital_ID column)

State column format: SA
Also contains Lat and Long columns for each hospital.

### public_hospital_list.csv
#### Dataset:

In [24]:
data = pd.read_csv(os.path.join(datadir, "public_hospital_list.csv"))
print(data.shape)
data.head()

(696, 19)


State              Hospital name Establishment ID Medicare Provider No.  \
0   ACT  Calvary Health Care Bruce        810000083              3531230T   
1   ACT          Canberra Hospital        810000082              0090030B   
2   ACT         QEII Family Centre        810000084              0090020F   
3   NSW                   Armidale        1180J2010              0010570T   
4   NSW                     Auburn        1174D2010              0010020A   

   Local Hospital Network identifier        Local Hospital Network  \
0                                801  Australian Capital Territory   
1                                801  Australian Capital Territory   
2                                801  Australian Capital Territory   
3                                109            Hunter New England   
4                                104                Western Sydney   

                                      Address Line 1  \
0  1 Hayden                                      ...   
1   Yamba                                         Dr   
2  129 Carruthers                                ...   
3  226 Rusden                                    ...   
4  18 Hargrave                                   ...   

                                      Address Line 2  asgc_ra Remoteness area  \
0  Bruce                                         ...        0    Major Cities   
1  Garran                                        ...        0    Major Cities   
2  Curtin                                        ...        0    Major Cities   
3  Armidale                                      ...        1  Inner Regional   
4  Auburn                                        ...        0    Major Cities   

   Number of available beds 2012-13 Peer Group code          Peer Group Name  \
0                     303.0                       L     Public acute group A   
1                     781.0                      PR       Principal referral   
2                      26.0                      O1  Early parenting centres   
3                      94.0                       M     Public acute group B   
4                     124.2                       M     Public acute group B   

  Provided data for NPHED Provided data for NHMD Provided data for NAPEDCD  \
0                     Yes                    Yes                       Yes   
1                     Yes                    Yes                       Yes   
2                     Yes                     No                        No   
3                     Yes                    Yes                       Yes   
4                     Yes                    Yes                       Yes   

  Provided data for ESWT Provided data for NNAPCD  \
0                    Yes                      Yes   
1                    Yes                      Yes   
2                     No                       No   
3                    Yes                      Yes   
4                    Yes                      Yes   

         IHPA funding designation  
0  Activity-based funded hospital  
1  Activity-based funded hospital  
2           Block-funded hospital  
3  Activity-based funded hospital  
4  Activity-based funded hospital

#### Notes:
A lot of metadata for hospitals in Australia. Contains hospital name, address, phone number, and website.

### statistic_id608819_population-distribution-australia-2021-by-state.xlsx
#### Dataset:

In [12]:
file = os.path.join(datadir, 'statistic_id608819_population-distribution-australia-2021-by-state.xlsx')

# Get sheet names
sheets = pd.ExcelFile(file).sheet_names

tables = {}
for sheet in sheets:
    tables[sheet] = pd.read_excel(file, sheet_name=sheet)

for sheet, table in tables.items():
    print(sheet)
    print(table.head())

Overview
   Unnamed: 0                                         Unnamed: 1 Unnamed: 2  \
0         NaN                                                NaN        NaN   
1         NaN                       Statistic as Excel data file        NaN   
2         NaN  Population distribution in Australia in 2021, ...        NaN   
3         NaN                                        Access data        NaN   
4         NaN                                                NaN        NaN   

   Unnamed: 3 Unnamed: 4  
0         NaN        NaN  
1         NaN        NaN  
2         NaN        NaN  
3         NaN        NaN  
4         NaN        NaN  
Data
   Unnamed: 0                                         Unnamed: 1  Unnamed: 2  \
0         NaN                                                NaN         NaN   
1         NaN  Population distribution Australia 2021, by state          NaN   
2         NaN  Population distribution in Australia in 2021, ...         NaN   
3         NaN                

#### Notes:
Not that useful, has a metadata sheet and Data sheet with population data for each state in Australia. (% values per state)
Maybe useful to determine the importance of each state in terms of accessiblity to hospitals.

### mb_2016_centroids.csv
#### Dataset:

In [13]:
file = os.path.join(datadir, 'mb_2016_centroids.csv')

data = pd.read_csv(file)
data.head()

MB_CODE16          MB_CAT16   SA1_MAIN16  SA1_7DIG16  SA2_MAIN16  \
0  80000009499  NOUSUALRESIDENCE  89999949999     8949999   899999499   
1  80000010000          Parkland  80105112301     8112301   801051123   
2  80000020000          Parkland  80105112301     8112301   801051123   
3  80000030000          Parkland  80105104904     8104904   801051049   
4  80000040000          Parkland  80105104903     8104903   801051049   

   SA2_5DIG16              SA2_NAME16  SA3_CODE16              SA3_NAME16  \
0       89499  No usual address (ACT)       89999  No usual address (ACT)   
1       81123          Black Mountain       80105          North Canberra   
2       81123          Black Mountain       80105          North Canberra   
3       81049                   Acton       80105          North Canberra   
4       81049                   Acton       80105          North Canberra   

   SA4_CODE16                    SA4_NAME16 GCC_CODE16  \
0         899        No usual address (ACT)      89499   
1         801  Australian Capital Territory      8ACTE   
2         801  Australian Capital Territory      8ACTE   
3         801  Australian Capital Territory      8ACTE   
4         801  Australian Capital Territory      8ACTE   

                     GCC_NAME16  STE_CODE16                    STE_NAME16  \
0        No usual address (ACT)           8  Australian Capital Territory   
1  Australian Capital Territory           8  Australian Capital Territory   
2  Australian Capital Territory           8  Australian Capital Territory   
3  Australian Capital Territory           8  Australian Capital Territory   
4  Australian Capital Territory           8  Australian Capital Territory   

   AREASQKM16        layer      xcoord     ycoord  
0      0.0000  MB_2016_ACT         NaN        NaN  
1      0.6714  MB_2016_ACT  149.082276 -35.291848  
2      5.8873  MB_2016_ACT  149.098226 -35.272163  
3      0.3145  MB_2016_ACT  149.101801 -35.288490  
4      0.1511  MB_2016_ACT  149.119356 -35.292336

#### Notes:
Contains the centroids of each Mesh Block in Australia. (MB_CODE16 column)
Also has various codes (SA1, SA2, SA3, SA4, GCC). (SA1_MAIN16, SA2_5DIG16, SA2_MAIN16, SA2_5DIG16)

### australia-latest.osm.pbf
#### Dataset:

In [17]:
file = os.path.join(datadir, 'australia-latest.osm.pbf')

import geopandas as gpd

# VERY SLOW, MAYBE USEFUL LATER:
# gdf = gpd.read_file(file, driver='OSM')
# gdf.head()

#### Notes:
This is a map of Australia in OpenStreetMap format. It is a binary file and can be read using the osmium or geopandas library in Python. It contains information about the roads, buildings, and other features in Australia.

## Processing plan
### RQ1: Healthcare Accessibility in Relation to Population Density
#### Question
How does the accessibility of healthcare facilities vary across regions with different population densities?
#### Context
Using data on hospital locations, travel distances, and population
density, this question explores whether healthcare facilities are fairly distributed and
accessible.

#### Interesting datasets:
- `ABS_Population_and_people_by_2021_SA2_(Beta).csv`<br>
Data on population density for each SA2 region in Australia.
- `duration_sa2_hospitals.pkl`<br>
Data on travel times between SA2 regions and hospitals.
- `duration_mb_hospitals.pkl`<br>
Data on travel times between Mesh Blocks and hospitals.
- `myhospitals-contact-details.csv`<br>
Metadata for hospitals in Australia. (Contains hospital ID used in duration_sa2_hospitals.pkl and duration_mb_hospitals.pkl)
- `public_hospital_list.csv`<br>
Metadata for hospitals in Australia.
- `mb_2016_centroids.csv`<br>
Centroids of each Mesh Block in Australia.

### RQ2: Impact of Transportation Distance/Time on Hospital Utilization

#### Question
What is the relationship between transportation distance/time and hospital admissions or
inpatient days?

#### Context
This question examines whether transportation time/distance affects hospital
utilization rates, potentially providing insights into resource allocation and healthcare
delivery.

#### Interesting datasets:
- `duration_sa2_hospitals.pkl`<br>
Data on travel times between SA2 regions and hospitals.
- `duration_mb_hospitals.pkl`<br>
Data on travel times between Mesh Blocks and hospitals.
- `ed_admissions.csv`<br>
Data on emergency department admissions for each SA2 region in Australia.
- `myhospitals-contact-details.csv`<br>
Metadata for hospitals in Australia. (Contains hospital ID used in duration_sa2_hospitals.pkl and duration_mb_hospitals.pkl)
- `public_hospital_list.csv`<br>
Metadata for hospitals in Australia.

## Additional analysis
### Hospital metadata
Join the hospital metadata datasets to get a complete list of hospitals in Australia. See if rows are lost in the process.

In [31]:
hospital_details = pd.read_csv(os.path.join(datadir, "myhospitals-contact-details.csv"), encoding="latin1")
hospital_list = pd.read_csv(os.path.join(datadir, "public_hospital_list.csv"))

# Join the two tables
hospital_data = hospital_details.merge(hospital_list, on="Hospital name")
hospital_data.head()

Index(['Hospital name', 'Phone number', 'Street address', 'Suburb', 'Postcode',
       'State_x', 'Local Hospital Network (LHN)',
       'Primary Health Network area (PHN)', 'Website', 'Description', 'Sector',
       'Beds', 'Latitude', 'Longitude', 'Hospital_ID', 'State_y',
       'Establishment ID', 'Medicare Provider No.',
       'Local Hospital Network identifier', 'Local Hospital Network',
       'Address Line 1', 'Address Line 2', 'asgc_ra', 'Remoteness area',
       'Number of available beds', '2012-13 Peer Group code',
       'Peer Group Name', 'Provided data for NPHED', 'Provided data for NHMD',
       'Provided data for NAPEDCD', 'Provided data for ESWT',
       'Provided data for NNAPCD', 'IHPA funding designation'],
      dtype='object')

#### Notes:
Nice! The hospital metadata datasets have different columns, so we can join them to get a complete list of hospitals in Australia. (e.g. first dataset has `Beds` column, second dataset has `Number of available beds` column, this can be useful for further analysis)